### ISSUE TITLE
[Bug]: docker install MO 0.8.0 failed

### ISSUE THREAD

**Issue by dengn** [matrixorigin/matrixone#10593]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 0.8.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

original issue: https://github.com/matrixorigin/matrixorigin.io.cn/issues/508

I've also recreated this issue in a centos 7.9 and docker version 1.13. The developer who fired this issue used docker version 24.0.4. 

But this docker problem doesn't happen in debian 11.1 and macOS. 

It seems to be a permission related issue.

**Comment by dengn** [comment]
with --privileged=true it is able to run now. 
```
docker run -d -p 6001:6001 --name matrixone --privileged=true matrixorigin/matrixone:0.8.0
```

**Comment by loveRhythm1990** [comment]
related with: https://github.com/docker-library/golang/issues/467

the best option maybe disable CGO. also can be fixed by use --priviledge option, we need add document to explain this before CGO is disabled

**Comment by dengn** [comment]
doc has been updated with this --priviledge option. Let's say this is a temporary fix for now. We may search for better solutions. 

https://docs.matrixorigin.cn/0.8.0/MatrixOne/Get-Started/install-on-linux/install-on-linux-method3/#1-docker

<img width="756" alt="image" src="https://github.com/matrixorigin/matrixone/assets/4965857/089c456d-1cd5-4024-ba9c-e7a7828a1404">

But this is still a problem, users may still have doubt or don't have this privilege at all. As I know, for windows, this is a super painful operation.

**Comment by loveRhythm1990** [comment]
need check if mo support disable CGO

**Comment by loveRhythm1990** [comment]
in progress

**Comment by loveRhythm1990** [comment]
will check this issuer in next week

**Comment by loveRhythm1990** [comment]
sorry for delay, will check it in next few days

**Comment by loveRhythm1990** [comment]
I  am working on it now

**Comment by loveRhythm1990** [comment]
this seems work perfectly on centos 7 and docker 24.0.5, i will try to reproduce it on docker 24.0.4
<img width="1731" alt="image" src="https://github.com/matrixorigin/matrixone/assets/52204121/0870517b-29e0-446c-bb18-2f1a73aa1d9b">

**Comment by loveRhythm1990** [comment]
I cannot reproduce this issue on following env:
1. centos 7, use image here: http://isoredirect.centos.org/centos/7/isos/x86_64/
2. docker version, I test version 24.0.5、24.0.4 and 18.06.3-ce,   but 1.13 is too old, I did not test it.

I install docker following steps here:  https://docs.docker.com/engine/install/centos/#install-using-the-repository
then I startup MO by:
```
docker run -d -p 6001:6001 --name matrixone registry.cn-shanghai.aliyuncs.com/matrixorigin/matrixone:0.8.0
```

do I miss some points to reproduce this issue @dengn

**Comment by loveRhythm1990** [comment]
in following env, Mo start successfully and run normally

tencent cloud, 
docker 24.0.5
centos kernel version: 4.18.0-348.7.1.el8_5.x86_64
 
<img width="1396" alt="image" src="https://github.com/matrixorigin/matrixone/assets/52204121/d507ed9a-cff5-4b41-a05a-06d808255a07">

**Comment by loveRhythm1990** [comment]
After communicating with person who open original issue, it turns out that the cause of this issue is version of docker client and docker server mismatched: the docker client version is 24.0.4 while the server version is 20.10.9.  after upgrade docker server to 24.0.4 the probelm is solved.

The reason behind this maybe the 24.0.4 client may enable more default [seccomp option](https://docs.docker.com/engine/security/seccomp/) while server not default allowed. 

we may need to determine a porper docker version scope for our user.

![image](https://github.com/matrixorigin/matrixone/assets/52204121/dc551f29-ded3-4874-b625-14c6e61a05d2)

**Comment by aylei** [comment]
@dengn @loveRhythm1990 

The root cause is go 1.20.0 introduce a new syscall which will blocked by the default seccompl profile in older versions of docker engine. The problem has been fixed in https://github.com/moby/moby/pull/43594, the fix is available in docker 23.x or 24.x version and has been back-ported to `20.10.18` (https://docs.docker.com/engine/release-notes/20.10/#201018)

We need to update the documentation to notice people that they should and `--privileged=true` if their docker version is lower than the above versions, or they can upgrade the docker version according to docker's doc.

**Comment by loveRhythm1990** [comment]
documents need update

**Comment by loveRhythm1990** [comment]
@dengn hi, should this issue be closed ? since there is another issue tracing document enhancement
